### ISSUE TITLE
 'ParallelOperationThreadCount' is larger than maximum of '64' on WP8.  v6.1.0

### ISSUE THREAD

**Issue by PaulDotNet** [Azure/azure-storage-net#208]
I encountered this problem couple years ago. Then it disappeared and now it is back in v6.1.0
The problem happens in CommonUtility.cs file in function AssertInBounds on line 193.
```
        internal static void AssertInBounds<T>(string paramName, T val, T min, T max)
            where T : IComparable
        {
            if (val.CompareTo(min) < 0)
            {
                throw new ArgumentOutOfRangeException(paramName, string.Format(CultureInfo.InvariantCulture, SR.ArgumentTooSmallError, paramName, min));
            }

            if (val.CompareTo(max) > 0) // <===== WATCH THIS!
            {
                throw new ArgumentOutOfRangeException(paramName, string.Format(CultureInfo.InvariantCulture, SR.ArgumentTooLargeError, paramName, max));
            }
        }
```
The line marked with WATCH THIS returns true when val is 1 and max is 64. It does not make any sense yet it happens when debugging on actual WP8.1 device

Just in case here is the full stack trace.
```
>	Microsoft.WindowsAzure.Storage.ni.DLL!Microsoft.WindowsAzure.Storage.Core.Util.CommonUtility.AssertInBounds<int>(string paramName, int val, int min, int max) Line 195	C#
 	Microsoft.WindowsAzure.Storage.ni.DLL!Microsoft.WindowsAzure.Storage.Blob.BlobRequestOptions.ParallelOperationThreadCount.set(int? value) Line 268	C#
 	Microsoft.WindowsAzure.Storage.ni.DLL!Microsoft.WindowsAzure.Storage.Blob.CloudBlobClient.CloudBlobClient(Microsoft.WindowsAzure.Storage.StorageUri storageUri, Microsoft.WindowsAzure.Storage.Auth.StorageCredentials credentials) Line 88	C#
 	Microsoft.WindowsAzure.Storage.ni.DLL!Microsoft.WindowsAzure.Storage.CloudStorageAccount.CreateCloudBlobClient() Line 578	C#
```

**Comment by mirobers** [comment]
How often do you see this and can you repro at will? Does it only happen while debugging?

**Comment by pemari-msft** [comment]
Sorry about that, @PaulDotNet. This is a bug in WP that we accidentally hit again in 6.1. We'll make a fix for the next release.

**Comment by PaulDotNet** [comment]
On WP emulator it does not happen.
On actual hardware in debugger this problem happens every time. 
On actual hardware in release mode it could be happening too as the app was crashing on that azure lib call when constructor was not in a catch block.
The same lib in WPF desktop app - no problems at all.
Paul

Date: Fri, 4 Dec 2015 09:53:57 -0800

**Comment by PaulDotNet** [comment]
Is this compiler problem? CompareTo returns -1 in that case but it should be signed integer. Seems like it is treated as unsigned for some reason. Do you know what is the root cause of this? 
Paul
Date: Fri, 4 Dec 2015 10:30:17 -0800

**Comment by pemari-msft** [comment]
Yes, it is an issue in the compiler.

**Comment by pemari-msft** [comment]
Fixed in version 6.2 of our library. Please let us know if you have further issues, thanks for the feedback!
